crates/compiler/parse/src/ast.rs

```diff
@@ -549,6 +591,50 @@ impl<'a> Defs<'a> {
         let tag = EitherIndex::from_left(type_def_index);
         self.push_def_help(tag, region, spaces_before, spaces_after)
     }
+
+    // Find the first definition that is a Apply Suffixed
+    // We need the tag_index so we can use it to remove the value
+    // We need the value index to know if it is the first
+    pub fn search_suffixed_defs(&self) -> Option<(usize, usize)> {
+        for (tag_index, tag) in self.tags.iter().enumerate() {
+            let index = match tag.split() {
+                Ok(_) => continue,
+                Err(value_index) => value_index.index(),
+            };
+
+            match &self.value_defs[index] {
+                ValueDef::Body(_, expr) => match expr.value {
+                    Expr::Apply(sub_expr, _, _) => match sub_expr.value {
+                        Expr::Suffixed(_) => {
+                            return Some((tag_index, index));
+                        }
+                        _ => continue,
+                    },
+                    _ => continue,
+                },
+                _ => continue,
+            }
+        }
+
+        None
+    }
+
+    // For desugaring Suffixed Defs we need to split the defs around the Suffixed value
+    pub fn split_values_either_side_of(&self, index: usize) -> (Self, Self) {
+        let mut before = self.clone();
+        let mut after = self.clone();
+
+        before.tags = self.tags[0..index].to_vec();
+
+        if index + 1 > self.tags.len() {
```

Comment:
I'd do `if index >= self.tags.len()` here, that way there's no chance of integer overflow causing problems.